lib/auth/bot.go

```diff
@@ -203,14 +216,22 @@ func (a *Server) updateBotInstance(
 	}
 
 	// Check if this bot instance actually exists.
+	var instanceGeneration int32
```

Comment:
One more TODO for the pile, we'll need to account for the upgrade case with a bot joining with an old-style counter inside a user label.